kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/activity/ConversationActivity.java

```diff
@@ -592,10 +593,41 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
                         final Uri singleFileUri = (data == null ? null : data.getData());
-                        kmAttachmentsController.processFile(singleFileUri, alCustomizationSettings, prePostUIMethods);
+                        int returnCode = kmAttachmentsController.processFile(singleFileUri, alCustomizationSettings, prePostUIMethods);
+                        doReturnCodeActions(returnCode);
                     } else {
-                        for (int i = 0; i < clipData.getItemCount(); i++) {
-                            kmAttachmentsController.processFile(clipData.getItemAt(i).getUri(), alCustomizationSettings, prePostUIMethods);
+                        int itemCount = clipData.getItemCount();
+                        if(itemCount > KmAttachmentsController.NO_OF_MULTI_SELECTIONS_ALLOWED) {
+                            KmCustomDialog.KmAlertDialog dialogFragmentSelectLimitCrossed = new KmCustomDialog.KmAlertDialog();
+
+                            Bundle bundleDialogFragmentDetails = new Bundle();
+                            bundleDialogFragmentDetails.putString(KmCustomDialog.KmAlertDialog.TITLE, getString(R.string.max_media_select_title));
+                            bundleDialogFragmentDetails.putString(KmCustomDialog.KmAlertDialog.MESSAGE, getString(R.string.max_media_select_message, KmAttachmentsController.NO_OF_MULTI_SELECTIONS_ALLOWED));
+                            bundleDialogFragmentDetails.putString(KmCustomDialog.KmAlertDialog.POSITIVE_BUTTON_TEXT, getString(R.string.send_max_limit_files, KmAttachmentsController.NO_OF_MULTI_SELECTIONS_ALLOWED));
+                            bundleDialogFragmentDetails.putString(KmCustomDialog.KmAlertDialog.NEGATIVE_BUTTON_TEXT, getString(R.string.reselect_files));
+
+                            dialogFragmentSelectLimitCrossed.setArguments(bundleDialogFragmentDetails);
+
+                            dialogFragmentSelectLimitCrossed.setKmDialogClickListener(new KmCustomDialog.KmAlertDialog.KmDialogClickListener() {
+                                @Override
+                                public void onClickNegativeButton(DialogInterface dialog, int id) {
+                                    processMultiSelectGallery();
+                                }
+
+                                @Override
+                                public void onClickPositiveButton(DialogInterface dialog, int id) {
+                                    for (int i = 0; i < KmAttachmentsController.NO_OF_MULTI_SELECTIONS_ALLOWED; i++) {
+                                        int returnCode = kmAttachmentsController.processFile(clipData.getItemAt(i).getUri(), alCustomizationSettings, prePostUIMethods);
+                                        doReturnCodeActions(returnCode);
+                                    }
+                                }
+                            });
+                            dialogFragmentSelectLimitCrossed.show(getFragmentManager(), "FragmentSelectLimitCrossed");
```

Comment:
Move this string to strings file as this is part of localisation

Reply:
This is just the fragment tag, won't be visible to the user. Do we still need to localize it?